File: core/repository/ast/go_ast_engine.py
from __future__ import annotations

import re
from typing import Any, Dict, List


def parse_go_ast(source: str, path: str = "") -> Dict[str, Any]:
    imports = [{"module": m.group(1), "kind": "go_import"} for m in re.finditer(r'import\s+"([^"]+)"', source)]
    nodes = [{"name": m.group(1), "kind": "func"} for m in re.finditer(r"func\s+(\w+)", source)]
    return {
        "language": "go",
        "path": path,
        "nodes": sorted(nodes, key=lambda item: item["name"])[:5000],
        "imports": sorted(imports, key=lambda item: item["module"])[:2000],
        "calls": [],
        "bounded": True,
    }

File: src/repository/ast/goAstEngine.ts
/**
 * Converted from Python: core/repository/ast/go_ast_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function parseGoAst(source: any, path: any = ""): any {
  var imports: any = py.iter(py.reFinditer("import\\s+\"([^\"]+)\"", source, "")).map((m: any) => ({"module": m.group(1), "kind": "go_import"}));
  var nodes: any = py.iter(py.reFinditer("func\\s+(\\w+)", source, "")).map((m: any) => ({"name": m.group(1), "kind": "func"}));
  return {"language": "go", "path": path, "nodes": py.slice(py.sorted(nodes, {key: ((item: any) => py.at(item, "name")) as (item: any) => any}), null, 5000), "imports": py.slice(py.sorted(imports, {key: ((item: any) => py.at(item, "module")) as (item: any) => any}), null, 2000), "calls": [], "bounded": true};
}
